File: hp-server/src/main/java/net/hserver/hplite/domian/entity/UserConfigEntity.java
package net.hserver.hplite.domian.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import net.hserver.hplite.message.ConnectType;

@Data
@TableName("user_config")
public class UserConfigEntity {

    @TableId(type = IdType.AUTO)
    private Integer id;
    /**
     * 当前key
     */
    private String configKey;

    /**
     * 用户KEY
     */
    private String deviceKey;

    /**
     * 套餐IP
     */
    private String serverIp;

    /**
     * 套餐端口
     */

    private Integer serverPort;

    /**
     * 本地IP
     */
    private String localIp;

    /**
     * 本地端口
     */
    private Integer localPort;

    /**
     * 穿透类型
     */
    private ConnectType connectType;

    /**
     * 备注
     */
    private String remarks;

    /**
     * 端口
     */
    private Integer port;

    /**
     * 域名
     */
    private String domain;

    /**
     * 状态
     */
    private String statusMsg;


    private ProxyVersion proxyVersion=ProxyVersion.NONE;


    /**
     * SSL证书Key
     */
    private String certificateKey;

    /**
     * 证书内容
     */
    private String certificateContent;

    /**
     * 只支持这两种
     */
    public  enum ProxyVersion{
        V1,V2,NONE
    }
}

File: hp-server-golang/hp-server-lib/entity/user_config_entity.go
package entity

import "hp-server-lib/bean"

type UserConfigEntity struct {
	/**
	 * 主键
	 */
	Id *int `json:"id" gorm:"primaryKey;autoIncrement"`
	/**
	 *当前key
	 */
	ConfigKey string `json:"configKey"`
	/**
	 * 用户KEY
	 */
	DeviceKey string `json:"deviceKey"`
	/**
	 * 套餐IP
	 */
	ServerIp string `json:"serverIp"`

	/**
	 * 套餐端口
	 */

	ServerPort *int `json:"serverPort"`

	/**
	 * 本地IP
	 */
	LocalIp string `json:"localIp"`

	/**
	 * 本地端口
	 */
	LocalPort *int `json:"localPort"`

	/**
	 * 穿透类型
	 */
	ConnectType *bean.ConnectType `json:"connectType"`

	/**
	 * 备注
	 */
	Remarks string `json:"remarks"`

	/**
	 * 端口
	 */
	Port *int `json:"port"`

	/**
	 * 域名
	 */
	Domain *string `json:"domain"`

	/**
	 * 状态
	 */
	StatusMsg string `json:"statusMsg"`

	/**
	 * 代理版本
	 */
	ProxyVersion bean.ProxyVersion `json:"proxyVersion"`

	/**
	 * SSL证书Key
	 */
	CertificateKey string `json:"certificateKey"`

	/**
	 * 证书内容
	 */
	CertificateContent string `json:"certificateContent"`

	/**
	 * 用户ID
	 */
	UserId *int `json:"userId"`
}

func (UserConfigEntity) TableName() string {
	return "user_config"
}
